### PR TITLE
Bootstrap Github Actions CI pipeline

### LINKED ISSUE #36: Setup some real CI for this project. That is what it is all about after all.
Need to have a CI script run through the various versions of AGP and Gradle so the known supported combination stays good.

### PR DESCRIPTION
**Pull request by plastiv**
Bootstrap `./gradlew build` CI pipeline based on github actions. Relates to https://github.com/trevjonez/composer-gradle-plugin/issues/36

There is no additional registration or configuration seems to be needed. Execution time is also decent. Currently its ~5min including starting android emulator https://github.com/plastiv/composer-gradle-plugin/runs/473061103?check_suite_focus=true

I have mostly ~~copy-pasted~~ inspired from https://github.com/slackhq/keeper/blob/master/.github/workflows/ci.yml

I haven't included gradle dependency or build cache optimization for now. As well as uploading snapshot or publishing tasks. That is something to consider later.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: CI
+
+# Trigger the workflow on push or pull request
```
Comment (trevjonez) [inline .github/workflows/ci.yml]: NIT: drop this comment, mostly redundant info.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    # Use macOS for emulator hardware acceleration
```
Comment (trevjonez) [inline .github/workflows/ci.yml:8]: but keep this one, was perfect for explaining the runs-on choice. 🎉

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1.3.0
+        with:
+          java-version: 1.8
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
```
Comment (trevjonez) [inline .github/workflows/ci.yml]: I am node ignorant, any real risk to having this as a dynamic version?

**Review by trevjonez — COMMENTED** [review]
Thank you for spending the time to figure this out!

Any idea if I can enable it in this PR or must it land in master to work?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (plastiv) [reply .github/workflows/ci.yml]: Don't have strong preference. I'm not node user. Removed.

**Comment by plastiv** [comment]
> Thank you for spending the time to figure this out!
> 
> Any idea if I can enable it in this PR or must it land in master to work?

Must be a security measure. Master is not necessary I believe. Can be merged to other branch first.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (trevjonez) [reply .github/workflows/ci.yml]: you didn't have to do that. w/e if we hit issues we can deal with it then.

**Review by trevjonez — APPROVED** [review]

**Pull request merged**